samples/prometheus-grafana/sample.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Capture and visualize metrics using Prometheus and Grafana
+component: Metrics
+---
+
+[Prometheus](https://prometheus.io) is a monitoring solution for storing time series data like metrics. [Grafana](https://grafana.com) allows to query and visualize the data stored in Prometheus (and other sources). This sample demonstrates how to capture NServiceBus metrics, storing these in Prometheus and visualizing this using Grafana.
+
+
+
+![Grafana NServiceBus processing time](grafana-graph.png)
+
+
+
+This sample reports the following metrics to Prometheus:
+
+ * Fetched messages per second 
+ * Failed messages per second
+ * Successful messages per second
+ * Critical time 
+ * Processing time
+
+
+
+ The sample also shows data like `Average message processing time in milliseconds` and `Maximum critical time in milliseconds`. For a detailed explanation of these metrics refer to the [metrics captured section in the metrics documentation](/nservicebus/operations/metrics.md#metrics-captured).
+
+
+## Prerequisites
+
+To run this sample, download and run both Prometheus and Grafana. This sample uses Prometheus 1.7.1 and Grafana 4.4.2.
+
+- [prometheus.io](https://prometheus.io)
+- [grafana.com](https://grafana.com)
+
+
+## Code overview
+
+The sample simulates messages load with a random 10% failure rate using the `LoadSimulator` class:
+
+snippet: prometheus-load-simulator
+
+
+## Capturing metric values
+
+The following nuget packages are required:
+
+- NServiceBus.Metrics
+- prometheus-net
+
+A Prometheus service is hosted inside an endpoint via the nuget package `prometheus-net`. The service enables Prometheus to scrape data gathered by the metrics package. In the sample the service that exposes the data to scrape is hosted on `http://localhost:3030'. The service is started and stopped inside a feature startup task as shown below
+
+snippet: prometheus-flush-probe
+
+
+Custom observers need to be registered for the metric probes provided via `NServiceBus.Metrics`. This is all setup in the `PrometheusFeature`
+
+
+snippet: prometheus-enable-nsb-metrics
+
+
+The names provided by the `NServiceBus.Metrics` probes are not compatible with Prometheus. Thus the names need to be aligned with the [naming conventions defined by Prometheus](https://prometheus.io/docs/practices/naming/) by mapping them accordingly
```

Comment:
I think *thus* should not be used to start a sentence.

The NServiceBus.Metrics names need to be .....